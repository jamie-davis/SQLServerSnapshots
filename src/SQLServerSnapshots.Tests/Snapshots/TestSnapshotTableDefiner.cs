// ReSharper disable UnusedType.Global

using SnapshotTests;
using SQLServerSnapshots.Schemas;
using SQLServerSnapshots.Snapshots;
using SQLServerSnapshots.Tests.Configuration;
using TestConsoleLib;
using TestConsoleLib.Testing;
using Xunit;

namespace SQLServerSnapshots.Tests.Snapshots
{
    public class TestSnapshotTableDefiner
    {
        private SchemaStructure _schema;

        private const string CreateRelatedTables = @"
CREATE SCHEMA Test;
GO
CREATE TABLE [Test].[A_Main]
(
    MainId INT PRIMARY KEY IDENTITY,
    Name VARCHAR(100) NULL,
    CreatedDate DATETIME NOT NULL DEFAULT(GETDATE())
);
GO
CREATE TABLE [Test].[B_Related]
(
    BId INT PRIMARY KEY IDENTITY,
    Name VARCHAR(100) NULL,
    MainId INT NOT NULL,
CONSTRAINT fk_B_MainId
    FOREIGN KEY (MainId)
    REFERENCES [Test].[A_Main] (MainId)
);
GO
";
    
        public TestSnapshotTableDefiner()
        {
            DbController.TearDown();
            DbController.Run(CreateRelatedTables);
            _schema = SchemaStructureLoader.Load(DbController.Server, DbController.TestDbName, "Test");
        }

        [Fact]
        public void BasicTableStructureIsLoaded()
        {
            //Arrange
            var collection = new SnapshotCollection();

            //Act
            SnapshotTableDefiner.Define(collection, _schema);

            //Assert
            var output = new Output();
            collection.GetSchemaReport(output);
            output.Report.Verify();
        }
    }
}