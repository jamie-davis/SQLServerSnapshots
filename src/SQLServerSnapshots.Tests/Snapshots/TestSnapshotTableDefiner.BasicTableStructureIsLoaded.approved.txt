Table Definitions                                                                              
-----------------                                                                              
                   Exclude               Defined                                               
                   From       Include In By                                                    
Table Name         Comparison Comparison Types                                                 
------------------ ---------- ---------- -------                                               
[Test].[A_Main]    False      False                                                            
                                                                                               
           Is      Is                                                                      Referenced                     
           Primary Compare Is            Is          Is       Is       Is       Referenced Property   Sort      Sort  Date
    Name   Key     Key     Unpredictable Predictable Required Included Excluded Table Name Name       Direction Index Type
    ------ ------- ------- ------------- ----------- -------- -------- -------- ---------- ---------- --------- ----- ----
    MainId True    False   True          False       False    False    False                                              
                                                                                               
                   Exclude               Defined                                               
                   From       Include In By                                                    
Table Name         Comparison Comparison Types                                                 
------------------ ---------- ---------- -------                                               
[Test].[B_Related] False      False                                                            
                                                                                               
           Is      Is                                                                      Referenced                     
           Primary Compare Is            Is          Is       Is       Is       Referenced Property   Sort      Sort  Date
    Name   Key     Key     Unpredictable Predictable Required Included Excluded Table Name Name       Direction Index Type
    ------ ------- ------- ------------- ----------- -------- -------- -------- ---------- ---------- --------- ----- ----
    BId    True    False   True          False       False    False    False                                              
    MainId False   False   False         False       False    False    False    [Test].    MainId                         
                                                                                [A_Main]                                  
                                                                                               
                                                                                               